src/wiseagents/core.py

```diff
@@ -665,45 +952,133 @@ class WiseAgentRegistry:
     contexts : dict[str, WiseAgentContext] = {}
     tools: dict[str, WiseAgentTool] = {}
     
+    config: dict[str, Any] = {}
+    
+    redis_db : redis.Redis = None
+    
+    @classmethod
+    def find_file(cls, file_name, config_directory="") -> str:
+        """
+        Find the file in the current directory or the home directory.
+        """
+        # Step 1: Check the current directory
+        if os.path.isfile(file_name):
+            return file_name
+        
+        # Step 2: Check the home directory
+        home_dir = os.path.expanduser("~")
+        home_path = os.path.join(home_dir, config_directory,file_name)
+        if os.path.isfile(home_path):
+            return home_path
+        
+        # If the file is not found in any of these locations, return None
+        raise FileNotFoundError(f"File '{file_name}' not found in current directory, home directory, as '{config_directory}'/{file_name}.")
+    
+    @classmethod
+    def get_config(cls) -> dict[str, Any]:
+        """
+        Get the configuration and initialize the redis database
+        for more information see 
+        https://wise-agents.github.io/wise_agents_architecture/#distributed-architecture
+        """
+        try: 
+            if cls.config is None or cls.config == {}:
+                cls.find_file(file_name="registry_config.yaml", config_directory=".wise-agents")
+                file_name = cls.find_file(file_name="registry_config.yaml", config_directory=".wise-agents")
+                cls.config : Dict[str, Any] = yaml.load(open(file_name), Loader=yaml.FullLoader)
+            if cls.config.get("use_redis") == True and cls.redis_db is None:
+                if (cls.config.get("redis_ssl") is True):
+                    cls.redis_db = redis.Redis(
+                    host=cls.config["redis_host"], port=cls.config["redis_port"],
+                    username=cls.config["redis_username"], # use your Redis user. More info https://redis.io/docs/latest/operate/oss_and_stack/management/security/acl/
+                    password=cls.config["redis_password"], # use your Redis password
+                    ssl=True,
+                    ssl_certfile=cls.config["redis_ssl_certfile"],
+                    ssl_keyfile=cls.config["redis_ssl_keyfile"],
+                    ssl_ca_certs=cls.config["redis_ssl_ca_certs"])
+
+                else:
+                    cls.redis_db = redis.Redis(host=cls.config["redis_host"], port=cls.config["redis_port"])
+            return cls.config
+        except Exception as e:
+            logging.error(e)
+            exit(1)
+    
     @classmethod
     def register_agent(cls,agent : WiseAgent):
         """
         Register an agent with the registry
         """
-        cls.agents[agent.name] = agent
+        if (cls.get_config().get("use_redis") == True):
```

Comment:
Can we use self._use_redis here?

Reply:
Ah, just realized, this method is actually part of the registry class. The _use_redis variable is part of the context.